.github/workflows/py_main.yml

```diff
@@ -25,7 +25,7 @@ jobs:
         with:
           fetch-depth: 0
       - name: Install poetry
-        run: pipx install poetry
+        run: pipx install poetry==2.1.1
```

Comment:
what's the argument for pinning to this version in the CI? (I understand pinning, but why this version specifically?)

Reply:
I thought that is the latest. Not needed though.

Reply:
gotcha, I would personally leave it unpinned so the CI catches any version update issues as they come. But its not a sticking point for me